content/en/docs/components/pipelines/v2/compile-a-pipeline.md

```diff
@@ -1,104 +1,75 @@
 
-@dsl.component
-def addition_component(num1: int, num2: int) -> int:
-    return num1 + num2
+1.  Define a simple pipeline:
 
-@dsl.pipeline(name='addition-pipeline')
-def my_pipeline(a: int, b: int, c: int = 10):
-    add_task_1 = addition_component(num1=a, num2=b)
-    add_task_2 = addition_component(num1=add_task_1.output, num2=c)
-```
+    ```python
+    from kfp import compiler
+    from kfp import dsl
 
-Now we can compile the pipeline to the file `my_pipeline.yaml`:
+    @dsl.component
+    def addition_component(num1: int, num2: int) -> int:
+      return num1 + num2
 
-```python
-cmplr = compiler.Compiler()
-cmplr.compile(my_pipeline, package_path='my_pipeline.yaml')
-```
+    @dsl.pipeline(name='addition-pipeline')
+    def my_pipeline(a: int, b: int, c: int = 10):
+      add_task_1 = addition_component(num1=a, num2=b)
+      add_task_2 = addition_component(num1=add_task_1.output, num2=c)
+    ```
 
-Just as a pipeline is a template for a multi-step workflow, a component is a template for a single-step workflow. We can also compile the component `addition_component` directly:
+1.  Compile the pipeline to the file `my_pipeline.yaml`:
 
-```python
-cmplr.compile(addition_component, package_path='addition_component.yaml')
-```
+    ```python
+    cmplr = compiler.Compiler()
+    cmplr.compile(my_pipeline, package_path='my_pipeline.yaml')
+    ```
 
-The `Compiler.compile` method accepts a few optional additional parameters:
+1.  _Optional:_ Directly  compile the component `addition_component` to the file `addition_component.yaml`:
 
-**`pipeline_name`** (`string`)
+    ```python
+    cmplr.compile(addition_component, package_path='addition_component.yaml')
+    ```
 
-Sets the name of the pipeline (or component). This is written to IR as the  `pipelineInfo.name` field. Will override the `name` passed to the `@dsl.pipeline` decorator.
+<!-- TODO: Include pipeline or component name in the table #} -->
 
-The pipeline name, whether set through the decorator or the compiler, names your pipeline template. When you upload your pipeline, a pipeline context by this name will be created. The pipeline context enables the backend and the Dashboard to associate artifacts and executions created from runs of the same pipeline template. This allows you, for example, to compare metrics artifacts from multiple runs of the same training pipeline to find the best model.
+The `Compiler.compile` method accepts the following parameters:
 
-**`pipeline_parameters`** (`Dict[str, Any]`)
-
-A map of parameter names to argument values. This amounts to providing default values for pipeline or component parameters. These defaults can be overriden at pipeline submission time.
-
-**`type_check`** (`bool`)
-
-Whether to enable static type checking during compilation. For more information about type checking, see [Component I/O: Component interfaces and type checking][type-checking].
+| Name | Type | Description |
```

Comment:
In general, this table should be consistent with the parameter list here: https://kubeflow-pipelines.readthedocs.io/en/2.0.0b6/source/compiler.html#kfp.compiler.Compiler.compile-parameters

Let's either find the best description and make them consistent between the two locations or simply refer to the reference docs and not include this table.

I think I prefer the former option, but open to either.

Reply:
I would cross-reference to the table.

Reply:
Mostly accepted, but I personally feel that this table has some additional information that's useful for the user. I've tried to make the wording consistent between the two locations otherwise.